### PR TITLE
Update TTS Collection for 1.0

### PR DESCRIPTION
**Pull request by blisc**
- Deprecates use of torch_stft and associated parameters: stft_conv,  stft_exact_pad for the FilterbankFeatures and AudioToMelSpectrogramPreprocessor classes
- stft_exact_pad has been replaced with exact_pad
- Update AudioToMelSpectrogramPreprocessor's calculation of seq_len
- Remove yaml anchors from TTS yamls in favour of Omegaconf/Hydra anchors
- Remove add_misc and load_audio from AudioTextDataset and child classes
- Update tts models to adhere to fmax=8000 for LJSpeech and all other associated hyper parameters
- Change GlowVocoder's denoise to use torch's stft and istft

### CODE REVIEW

**Comment by blisc** [comment]
@titu1994 Can you take a look at the asr datalayer and feature preprocessor changes?

**Review by titu1994 — COMMENTED** [review]
Everything looks alright, and should mostly be backward compatible to some degree, plus thanks for adding those tests. 

Minor comments about removal of type info (sample rate) and the "add_misc" flag. I don't think anyone uses it, but perhaps we should do key checks and simply add it into the loaded manifest ? If @nithinraok doesn't use it for speaker, it's fine to remove it entirely.

File: nemo/collections/asr/data/audio_to_text.py [hunk 1]
```diff
@@ -99,21 +93,14 @@ class _AudioTextDataset(Dataset):
...
         """Returns definitions of module output ports.
                """
         return {
-            'audio_signal': NeuralType(
-                ('B', 'T'),
-                AudioSignal(freq=self._sample_rate)  # TODO: self._sample_rate is not defined anywhere
-                if self is not None and hasattr(self, '_sample_rate')
-                else AudioSignal(),
-            ),
+            'audio_signal': NeuralType(('B', 'T'), AudioSignal()),
```
[the remark this answers is not in the document]
Comment (blisc) [reply nemo/collections/asr/data/audio_to_text.py:103]: Because self._sample_rate is never defined, so it is always None.

[the remark this answers is not in the document]
Comment (titu1994) [reply nemo/collections/asr/data/audio_to_text.py:103]: Wow :D alright let's remove.

File: nemo/collections/asr/data/audio_to_text.py [hunk 2]
```diff
@@ -179,13 +159,6 @@ def __getitem__(self, index):
 
         output = f, fl, torch.tensor(t).long(), torch.tensor(tl).long()
 
-        if self._add_misc:
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/asr/data/audio_to_text.py:182]: I use label for speaker class. And also I use functions from audio_to_label.py not audio_to_text.py

[the remark this answers is not in the document]
Comment (titu1994) [reply nemo/collections/asr/data/audio_to_text.py:182]: Perfect. Safe to drop then.

**Comment by Oktai15** [comment]
@blisc I use misc as ability to match sample from dataloader and file name from manifest.

also TalkNet uses load_audio=False flag. 

I suggest to discuss before removing it.

**Comment by blisc** [comment]
> @blisc I use misc as ability to match sample from dataloader and file name from manifest.
> 
> also TalkNet uses load_audio=False flag.
> 
> I suggest to discuss before removing it.

Seeing as how only TalkNet seems to use it and TalkNet already has it's own datalayer at https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/data/audio_to_text.py#L291. I will continue to remove these from the base class. Feel free to add such functionality to the derived class instead.

**Comment by Oktai15** [comment]
@blisc 

It is not a problem to implement it in derived class, but why do you necessary want to remove it from base class? Maybe wii be another case for using load_audio=False.
